live-build/misc/live-build-hooks/81-upgrade-repository.binary

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -eux
+#
+# Copyright 2018 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This script is intended to be used as part of Delphix's build process.
+# It's role is to generate a tar archive that contains an Aptly/APT
+# repository of all of the packages that have been installed in the
+# live-build "binary" directory. The intention, is this archive can then
+# be used to upgrade an existing appliance, using the packages installed
+# in this particular variant build.
+#
+
+set -o pipefail
+
+#
+# Here, we generate a list of all of the packages that have been
+# installed in the "binary" chroot directory, including package version,
+# and then use this list to download each of these packages.
+#
+# We must perform the package download while chrooted in the binary
+# directory to ensure we use the same APT sources that were used when
+# installing the packages.
+#
+# This directory of download packages will later be passed to Aptly to
+# build our "upgrade repository".
+#
+chroot binary bash -eux <<'EOF'
+set -o pipefail
+
+mkdir -p /packages
```

Comment:
Do we need to worry about failure here?

Reply:
We're using `-eux` options to bash, so if it fails the script will fail, and the calling script will fail as well since that's also using `-eux`.. on any command failure, the whole build should end up failing.